subsys/dfu/img_util/flash_img.c

```diff
@@ -19,14 +20,11 @@
 #endif
 
 #include <zephyr/devicetree.h>
-#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
-	#define UPLOAD_FLASH_AREA_LABEL slot1_ns_partition
-#else
-#if FIXED_PARTITION_EXISTS(slot1_partition)
-	#define UPLOAD_FLASH_AREA_LABEL slot1_partition
+
+#if ZEPHYR_MCUBOOT_APP_0_SECONDARY_SLOT_EXISTS
+	#define UPLOAD_FLASH_AREA_LABEL ZEPHYR_MCUBOOT_APP_0_SECONDARY_SLOT_LABEL
 #else
-	#define UPLOAD_FLASH_AREA_LABEL slot0_partition
-#endif
+	#define UPLOAD_FLASH_AREA_LABEL ZEPHYR_MCUBOOT_APP_0_PRIMARY_SLOT_LABEL
```

Comment:
Would need to think about this for direct XIP mode also, if the current slot running is the secondary slot, it needs to write to the first slot, not the second

Reply:
Yes. This is problem we have with DirectXIP - for which slot app has been build and to which slot will it write.
In case of DirectXIP the primary and secondary slots are just names for two equivalent slots of some application.